source/AI.cpp

```diff
@@ -621,19 +649,33 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 
 		// Pick a target and automatically fire weapons.
 		shared_ptr<Ship> target = it->GetTargetShip();
+		// Find a targeted asteroid but only if the ship is player-owned.
+		shared_ptr<Minable> targetAsteroid = FindTargetAsteroid(*it);
 		if(isPresent && !personality.IsSwarming())
 		{
 			// Each ship only switches targets twice a second, so that it can
 			// focus on damaging one particular ship.
 			targetTurn = (targetTurn + 1) & 31;
-			if(targetTurn == step || !target || target->IsDestroyed() || (target->IsDisabled()
-					&& personality.Disables()) || !target->IsTargetable())
+			if(targetTurn == step && targetAsteroid)
+				it->SetTargetAsteroid(targetAsteroid);
+			else if(targetTurn == step || !target || target->IsDestroyed()
+				|| (target->IsDisabled() && personality.Disables()) || !target->IsTargetable())
 				it->SetTargetShip(FindTarget(*it));
 		}
 		if(isPresent)
 		{
 			AimTurrets(*it, firingCommands, it->IsYours() ? opportunisticEscorts : personality.IsOpportunistic());
```

Comment:
This might be misunderstanding this code.  I assumed turrets would not fire if they do not aim.

These lines you're reviewing aren't code I added.  The lines are in master branch now so I think this feedback may be out of scope for this pull request.  If changing these lines is desirable I can open a separate pull request.

Reply:
I misunderstood your original statement.  After looking at the code I understand and will fix this.

Reply:
Fixed